core/types/block.libevm.go

```diff
@@ -43,7 +43,7 @@ func (h *Header) hooks() HeaderHooks {
 	return new(NOOPHeaderHooks)
 }
 
-func (e ExtraPayloads[HPtr, SA]) hooksFromHeader(h *Header) HeaderHooks {
+func (e ExtraPayloads[HPtr, BodyExtraPtr, SA]) hooksFromHeader(h *Header) HeaderHooks {
```

Comment:
`BodyExtraPtr` is used instead of `BPtr` because `BlockExtraPtr` is coming, and we need a way to differentiate the two B's without it being overly confusing